src/components/TodoApp/TodoApp.tsx

```diff
@@ -0,0 +1,231 @@
+  const [isClicked, setIsClicked] = useState(false);
+  const [isCompleted, setIsCompleted] = useState(false);
+  const [isBeingEdited, setIsBeingEdited] = useState(0);
+  const activeTodos = todos.filter(todo => !todo.completed);
+
+  useEffect(() => {
+    getTodos(userId)
+      .then(todosFromServer => setTodos(todosFromServer))
+      .catch(() => setErrorMessage(ErrorNames.LoadingError));
+  }, []);
+
+  const handleSortTypeChange = (sortedType: SortType) => {
+    setSortType(sortedType);
+  };
+
+  const handleTodoTitleChange = (
+    event: React.ChangeEvent<HTMLInputElement>,
+  ) => {
+    setTodoTitle(event.target.value);
+  };
+
+  const handleFormSubmit = async (
+    title: string,
+  ) => {
+    if (!title.trim().length) {
+      setErrorMessage(ErrorNames.TitleError);
+
+      return;
+    }
+
+    setTempTodo({
+      id: 0,
+      title,
+      userId,
+      completed: false,
+    });
+
+    setIsInputDisabled(true);
+
+    try {
+      const createdTodo = await createTodo({ title, completed: false, userId });
+
+      setTempTodo(null);
+      setTodoTitle('');
+      setTodos(currentTodos => [...currentTodos, createdTodo]);
+    } catch {
+      setErrorMessage(ErrorNames.AddError);
+    } finally {
+      setIsInputDisabled(false);
+    }
+  };
+
+  const handleDeleteTodo = async (todoId: number) => {
+    try {
+      setIsUpdatingTodoId(todoId);
+
+      await deleteTodo(todoId);
+      setTodos(currentTodos => currentTodos.filter(todo => todo.id !== todoId));
+    } catch {
+      setErrorMessage(ErrorNames.DeleteError);
+    } finally {
+      setIsUpdatingTodoId(0);
+    }
+  };
+
+  const deleteCompletedTodos = () => {
+    const completedTodos = todos.filter(todo => todo.completed);
+
+    completedTodos.forEach(currentTodo => {
+      handleDeleteTodo(currentTodo.id);
+    });
+  };
+
+  const handleUpdateTodo = async (
+    title = '',
+    todoId: number,
+    completed: boolean,
+  ) => {
+    try {
+      setIsUpdatingTodoId(todoId);
+
+      if (!title.length) {
+        handleDeleteTodo(todoId);
+
+        return;
+      }
+
+      const updatedTodo = await updateTodo({
+        title,
+        completed,
+        id: todoId,
+        userId,
+      });
+
+      setTodos(currentTodos => {
+        return currentTodos
+          .map(todo => (todo.id === todoId ? updatedTodo as Todo : todo));
+      });
+    } catch {
+      setErrorMessage(ErrorNames.UpdateError);
+    } finally {
+      setIsUpdatingTodoId(0);
+    }
+  };
+
+  const setAllTodosComplete = async () => {
+    const newCompletedValue = !isCompleted;
+
+    setIsUpdatingTodoId(todos.length);
+
+    try {
+      const updateTodoAtIndex = async (index: number) => {
+        const todo = todos[index];
+
+        if (index >= todos.length) {
+          return;
+        }
+
+        await handleUpdateTodo(todo.title, todo.id, newCompletedValue);
+        await updateTodoAtIndex(index + 1);
+      };
+
+      await updateTodoAtIndex(0);
```

Comment:
don't need to user 'await' here, because you have already 'await' in method updateTodoAtIndex ( 150 and 151 lines )